tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CreateOrderItemServiceTests.cs

```diff
@@ -21,5 +35,243 @@ public static void Constructors_VerifyGuardClauses()
+            result.Success.Should().BeFalse();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task CreateAsync_ValidAggregateValidation_ValidateResult(
+            [Frozen] Mock<ICreateOrderItemValidator> orderItemValidator,
+            Order order,
+            CatalogueItemId catalogueItemId,
+            CreateOrderItemModel model,
+            CreateOrderItemService service)
+        {
+            var aggregateValidationResult = new AggregateValidationResult();
+            aggregateValidationResult.AddValidationResult(new ValidationResult(), 0);
+
+            var catalogueItemType = Enum.Parse<CatalogueItemType>(model.CatalogueItemType, true);
+
+            orderItemValidator.Setup(o => o.Validate(order, model, catalogueItemType)).Returns(aggregateValidationResult);
+
+            var result = await service.CreateAsync(order, catalogueItemId, model);
+            result.Success.Should().BeTrue();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task CreateAsync_SetsEstimationPeriod(
+            [Frozen] ApplicationDbContext context,
+            [Frozen] Mock<ICreateOrderItemValidator> orderItemValidator,
+            [Frozen] Mock<IServiceRecipientService> serviceRecipientService,
+            Order order,
+            CatalogueItemId catalogueItemId,
+            CreateOrderItemModel model,
+            CreateOrderItemService service)
+        {
+            context.Order.Add(order);
+            await context.SaveChangesAsync();
+
+            var aggregateValidationResult = new AggregateValidationResult();
+            aggregateValidationResult.AddValidationResult(new ValidationResult(), 0);
+
+            var catalogueItemType = Enum.Parse<CatalogueItemType>(model.CatalogueItemType, true);
+
+            var provisioningType = Enum.Parse<ProvisioningType>(model.ProvisioningType, true);
+
+            var serviceRecipients = model.ServiceRecipients.Select(r => new ServiceRecipient(r.OdsCode, r.Name));
+            orderItemValidator.Setup(o => o.Validate(order, model, catalogueItemType)).Returns(aggregateValidationResult);
+            serviceRecipientService.Setup(r => r.AddOrUpdateServiceRecipients(It.IsAny<IEnumerable<ServiceRecipient>>()))
+                .ReturnsAsync(serviceRecipients.ToDictionary(r => r.OdsCode));
+
+            await service.CreateAsync(order, catalogueItemId, model);
+
+            var expectedEstimationPeriod = catalogueItemType.InferEstimationPeriod(
+                provisioningType,
+                OrderingEnums.ParseTimeUnit(model.EstimationPeriod));
+
+            var expectedOrderItem = context.Set<OrderItem>().First(o => o.OrderId.Equals(order.Id));
+
+            expectedOrderItem.EstimationPeriod.Should().Be(expectedEstimationPeriod);
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task CreateAsync_RemovedDefaultDeliveryDate(
```

Comment:
I think tweaking this name slightly to `CreateAsync_RemovesDefaultDeliveryDate` might be better.